File: c-009/main.rb
def getLastPair a, b
    return Math.sqrt(a ** 2 + b ** 2)
end

def isPythagorasPair a, b
    return a ** 2 + b ** 2 === getLastPair(a, b) ** 2
end

def pythagoreanTripletExplosion sumTotal
    a = 1
    loop do
        for b in 1..a do
            if isPythagorasPair(a, b) then
                c = getLastPair(a, b)
                if a + b + c == sumTotal then
                    return (a * b * c).to_i
                end
            end
        end
        a = a + 1
    end
end

puts pythagoreanTripletExplosion(1000)

File: c-009/main.js
function getLastPair(a, b) {
    return Math.sqrt(a ** 2 + b ** 2)
}

function isPythagorasPair(a, b) {
    return a ** 2 + b ** 2 === getLastPair(a, b) ** 2
}

function pythagoreanTripletExplosion(sumTotal) {
    for (let a = 1;; a++) {
        for (let b = 1; a > b ; b++) {
            if (isPythagorasPair(a, b)) {
                const c = getLastPair(a, b)
                if (a + b + c === sumTotal) {
                    return a * b * c
                }
            }
        }
    }
}

console.log(pythagoreanTripletExplosion(1000))
